### PR TITLE
Blockchainn basics updated

### PR DESCRIPTION
**Pull request by Lakshya18jul**

### CODE REVIEW

File: src/questions/blockchain_basics.yaml [hunk 1]
```diff
@@ -1,35 +1,1100 @@
-- # uuid Needs to be a random UUID - You can generate it from https://www.uuidgenerator.net/version4.  Use version 4
-  uuid: 1b8fbe3c-aeb2-4b7d-a3ef-3082bf07efcc
-  # Types can be MultipleChoice or SingleChoice
+- uuid: 617bfecc-f141-49c5-abdf-9716e8ea80d5
+  type: SingleChoice
+  content: Which of the following is not a characteristic of Blockchain Technology?
+  hint: Properties of Blockchain
+  explanation: Blockchain allows peer to peer transaction , it is decentrlised and permits efficient barter system.
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: remove extra space before ','.  Please do find replace and do it for all the text

File: src/questions/blockchain_basics.yaml [hunk 2]
```diff
@@ -1,35 +1,1100 @@
...
+    - content: Allows for an efficient barter mechanism
+      key: C
+      order: 2
+    - content: None of the above
+      key: D
+      order: 3
+- uuid: 922879d6-4e38-4173-880e-94487624c4d0
+  type: SingleChoice
+  content: Each Block of the Blockchain consists of which the following?
+  hint: Blockchain Basics
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Not sure if `Blockchain Basics` is a relevant hint

File: src/questions/blockchain_basics.yaml [hunk 3]
```diff
@@ -1,35 +1,1100 @@
...
+  explanation: Each Block contains a hash pointer to the next node , current transaction details , Timestamp.
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: No space before ','

File: src/questions/blockchain_basics.yaml [hunk 4]
```diff
@@ -1,35 +1,1100 @@
...
+  choices:
+    - content: True
+      key: A
+      order: 0
+    - content: False
+      key: B
+      order: 1    
+- uuid: 505ebda8-92ba-4922-bd0f-32ebd145b922
+  type: SingleChoice
+  content: Blocks in Blockchain as linked to each to each other .......
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: `linked to each to each other` not a proper english

File: src/questions/blockchain_basics.yaml [hunk 5]
```diff
@@ -1,35 +1,1100 @@
...
+    - content: Forward direction
+      key: B
+      order: 1
+    - content: Not connected
+      key: C
+      order: 2
+- uuid: a6ce806e-02dd-4195-b254-dfb5aa0092f3
+  type: MultipleChoice
+  content: Identify Features of Blockchain?
+  hint: Characteristics of Blockchain
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Dont thin'k `Characteristics of Blockchain` is a relevant Hint. You can add NoHint for this

File: src/questions/blockchain_basics.yaml [hunk 6]
```diff
@@ -1,35 +1,1100 @@
...
+    - content: Peer to Peer
+      key: B
+      order: 1
+    - content: Product to Product
+      key: C
+      order: 2
+    - content: Public key to Private Key
+      key: D
+      order: 3
+- uuid: 284d96ab-0336-4c2f-81cf-d6afe087d987
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml:155]: I think we can remove this question `In Blockchain nodes are not in a distributed system.` its too basic

File: src/questions/blockchain_basics.yaml [hunk 7]
```diff
@@ -1,35 +1,1100 @@
...
+  choices:
+    - content: True
+      key: A
+      order: 0
+    - content: False
+      key: B
+      order: 1
+- uuid: 120f4e3b-7bb6-4ea0-b860-7e7bfdcf5baa
+  type: SingleChoice
+  content: The term used for Blockchain splits?
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Give context of `split`

File: src/questions/blockchain_basics.yaml [hunk 8]
```diff
@@ -1,35 +1,1100 @@
...
+      key: C
+      order: 2
+    - content: Satoshi Nakamoto
+      key: D
+      order: 3
+- uuid: 3110ead9-fde0-43c0-8007-b1ba3a6781fd
+  type: SingleChoice
+  content: Like every digital asset Bitcoin has the problem of “double spend”?
+  hint: Double Spend means copying and reselling the same digital asset many number of times.
+  explanation: Every bitcoin has a unique hash pointer attached to it hence it becomes impossible to create a copy with a hash present in the system.
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Not sure if explanation is correct. The question also needs to be reworded.

It's an important question though https://www.investopedia.com/ask/answers/061915/how-does-block-chain-prevent-doublespending-bitcoins.asp

`The first transaction would be approved via the confirmation mechanism and then verified into the subsequent block. However, the second transaction would be recognized as invalid by the confirmation process and would not be verified. If both transactions are pulled from the pool for confirmation simultaneously, the transaction with the highest number of confirmations will be included in the blockchain, while the other one will be discarded.`

File: src/questions/blockchain_basics.yaml [hunk 9]
```diff
@@ -1,35 +1,1100 @@
...
+      key: C
+      order: 2
+    - content: Portability
+      key: D
+      order: 3
+- uuid: 99349427-2e10-4649-9962-6b26fac55f05
+  type: MultipleChoice
+  content: What are the benefits of Fiat currency over Bitcoin?
+  hint: Benefits of centralised system
+  explanation: Fiat currency has low transaction fees , they are backed by gold and thus possess a high tangible value.
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: US dollar is not backed by gold. We can say they are `mostly backed by gold`

File: src/questions/blockchain_basics.yaml [hunk 10]
```diff
@@ -1,35 +1,1100 @@
...
+  explanation: Proof of Work consensus mechanism is used to mine Bitcoin.
+  answerKeys:
+    - C
+  subTopics:
+    - "introduction-to-blockchain"
+  difficultyLevel: "Low"
+  choices:
+    - content: Proof of Authority
+      key: A
+      order: 0
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: we can remove all `order: ` as they are not needed

File: src/questions/blockchain_basics.yaml [hunk 11]
```diff
@@ -1,35 +1,1100 @@
...
+      key: C
+      order: 2
+    - content: Proof of History
+      key: D
+      order: 3
+- uuid: e9b6538b-a132-468e-8728-0818b9b3acba
+  type: SingleChoice
+  content: Bitcoin has got unlimited supply.
+  hint: NoHint
+  explanation: Bitcoin works on Proof of work mechanism.
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: List the maximum supply in the explanation

File: src/questions/blockchain_basics.yaml [hunk 12]
```diff
@@ -1,35 +1,1100 @@
...
+      order: 1
+    - content: Changpen Zhao
+      key: C
+      order: 2
+    - content: Brian Armstrong
+      key: D
+      order: 3
+- uuid: 6c7083e5-ae8f-407d-bdbd-14f8b5327d95
+  type: SingleChoice
+  content: Ethereum mining works on the principle of ......
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Ethereum mining works on the principle of ......

I think so far it was proof of work and now they are switching to proof of stake. So we need to reword the question

File: src/questions/blockchain_basics.yaml [hunk 13]
```diff
@@ -1,35 +1,1100 @@
...
+    - content: Open source
+      key: A
+      order: 0
+    - content: Smart contracts
+      key: B
+      order: 1
+    - content: Oracles
+      key: C
+      order: 2
+    - content: Consensus Mechanism
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml:491]: Is consensus mechanism not used in Bitcoin?

File: src/questions/blockchain_basics.yaml [hunk 14]
```diff
@@ -1,35 +1,1100 @@
...
+  choices:
+    - content: True
+      key: A
+      order: 0
+    - content: False
+      key: B
+      order: 1
+- uuid: 8bc6e329-f119-430c-84a0-01aaf01433fe
+  type: MultipleChoice
+  content: What is most common problem with simple hashes?
```
Comment (RobinNagpal) [inline src/questions/blockchain_basics.yaml]: Add more context to `simple`. Not tech people don't know what are simple and non-simple hashed

**Review by RobinNagpal — CHANGES_REQUESTED** [review]

File: src/questions/blockchain_basics.yaml (hunk 1 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 2 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Gave a more detailed hint

File: src/questions/blockchain_basics.yaml (hunk 4 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Removed the redundant portion

File: src/questions/blockchain_basics.yaml (hunk 5 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Updated

File: src/questions/blockchain_basics.yaml (hunk 6 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml:155]: Removed and added a new question

File: src/questions/blockchain_basics.yaml (hunk 7 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Provided the description

File: src/questions/blockchain_basics.yaml (hunk 8 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 9 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 10 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 11 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 12 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml]: Done

File: src/questions/blockchain_basics.yaml (hunk 13 above)
Comment (Lakshya18jul) [reply src/questions/blockchain_basics.yaml:491]: Done

**Pull request merged**